src/EditorFeatures/Core/Classification/TotalClassificationTaggerProvider.cs

```diff
@@ -0,0 +1,230 @@
+    {
+        var tagger = CreateTagger(textView, buffer);
+        if (tagger is not ITagger<T> typedTagger)
+        {
+            tagger?.Dispose();
+            return null;
+        }
+
+        return typedTagger;
+    }
+
+    public TotalClassificationAggregateTagger? CreateTagger(ITextView textView, ITextBuffer buffer)
+    {
+        var syntacticTagger = _syntacticTaggerProvider.CreateTagger(buffer);
+        var semanticTagger = _semanticTaggerProvider.CreateTagger(textView, buffer);
+        var embeddedTagger = _embeddedTaggerProvider.CreateTagger(textView, buffer);
+
+        if (syntacticTagger is null || semanticTagger is null || embeddedTagger is null)
+        {
+            syntacticTagger?.Dispose();
+            semanticTagger?.Dispose();
+            embeddedTagger?.Dispose();
+            return null;
+        }
+
+        return new TotalClassificationAggregateTagger(syntacticTagger, semanticTagger, embeddedTagger);
+    }
+
+    internal sealed class TotalClassificationAggregateTagger(
+        SyntacticClassificationTaggerProvider.Tagger syntacticTagger,
+        SimpleAggregateTagger<IClassificationTag> semanticTagger,
+        SimpleAggregateTagger<IClassificationTag> embeddedTagger)
+        : AbstractAggregateTagger<IClassificationTag>(ImmutableArray.Create<ITagger<IClassificationTag>>(syntacticTagger, semanticTagger, embeddedTagger))
+    {
+        private static readonly Comparison<ITagSpan<IClassificationTag>> s_spanComparison = static (s1, s2) => s1.Span.Start - s2.Span.Start;
+
+        public override SegmentedList<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        {
+            // First, get all the syntactic tags.  While they are generally overridden by semantic tags (since semantics
+            // allows us to understand better what things like identifiers mean), they do take precedence for certain
+            // tags like 'Comments' and 'Excluded Code'.  In those cases we want the classification to 'snap' instantly to
+            // the syntactic state, and we do not want things like semantic classifications showing up over that.
+
+            var totalTags = new SegmentedList<ITagSpan<IClassificationTag>>();
+
+            using var _ = Classifier.GetPooledList<ITagSpan<IClassificationTag>>(out var stringLiterals);
+
+            var syntacticSpans = syntacticTagger.GetTags(spans);
+            var semanticSpans = semanticTagger.GetTags(spans);
+
+            syntacticSpans.Sort(s_spanComparison);
+            semanticSpans.Sort(s_spanComparison);
+
+            var syntacticEnumerator = syntacticSpans.GetEnumerator();
+            var semanticEnumerator = semanticSpans.GetEnumerator();
+
+            var currentSyntactic = GetNextSyntacticSpan();
+            var currentSemantic = GetNextSemanticSpan();
+
+            while (currentSyntactic != null && currentSemantic != null)
```

Comment:
this follows a similar strategy/code we have when producing tags where we diff the old and new tags against each other.  because they're sorted, we can just walk them in order, taking from the earlier one to produce the final result.  one when we hit spns of interest do we have to look at the other and decide what to do.